orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pollers/EphemeralServerGroupsPoller.java

```diff
@@ -0,0 +1,249 @@
+import com.netflix.spinnaker.orca.pipeline.ExecutionLauncher;
+import com.netflix.spinnaker.orca.pipeline.model.Execution;
+import com.netflix.spinnaker.security.AuthenticatedRequest;
+import com.netflix.spinnaker.security.User;
+import groovy.util.logging.Slf4j;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.stereotype.Component;
+import redis.clients.jedis.Jedis;
+import redis.clients.util.Pool;
+import rx.Observable;
+
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static com.netflix.spinnaker.orca.clouddriver.tasks.servergroup.EphemeralServerGroupEntityTagGenerator.TTL_TAG;
+import static java.lang.String.format;
+
+@Slf4j
+@Component
+@ConditionalOnExpression(value = "${pollers.ephemeralServerGroups.enabled:false}")
+public class EphemeralServerGroupsPoller extends AbstractPollingNotificationAgent {
+  private static final Logger log = LoggerFactory.getLogger(EphemeralServerGroupsPoller.class);
+
+  private final ObjectMapper objectMapper;
+  private final OortService oortService;
+  private final RetrySupport retrySupport;
+  private final ExecutionLauncher executionLauncher;
+
+  private final PollerSupport pollerSupport;
+
+  private final Counter errorsCounter;
+  private final Counter triggeredCounter;
+
+  @Autowired
+  public EphemeralServerGroupsPoller(@Qualifier("jedisPool") Pool<Jedis> jedisPool,
+                                     ObjectMapper objectMapper,
+                                     OortService oortService,
+                                     RetrySupport retrySupport,
+                                     Registry registry,
+                                     ExecutionLauncher executionLauncher) {
+    super(jedisPool);
+
+    this.objectMapper = objectMapper;
+    this.oortService = oortService;
+    this.retrySupport = retrySupport;
+    this.executionLauncher = executionLauncher;
+
+    this.pollerSupport = new PollerSupport(objectMapper, retrySupport, oortService);
+
+    this.errorsCounter = registry.counter("poller.ephemeralServerGroups.errors");
+    this.triggeredCounter = registry.counter("poller.ephemeralServerGroups.triggered");
+  }
+
+  @Override
+  protected long getPollingInterval() {
+    return TimeUnit.MINUTES.toSeconds(10);
+  }
+
+  @Override
+  protected String getNotificationType() {
+    return "ephemeralServerGroups";
+  }
+
+  @Override
+  protected void startPolling() {
+    subscription = Observable
```

Comment:
wondering in general if we thing Rx is a useful tool or extra complexity in the spinnaker codebase

(vs just a ScheduledExecutorService and scheduleAtFixedRate)